File: PolicyPlus/DetailAdmx.vb
Public Class DetailAdmx
    Public Sub PresentDialog(Admx As AdmxFile, Workspace As AdmxBundle)
        TextPath.Text = Admx.SourceFile
        TextNamespace.Text = Admx.AdmxNamespace
        TextSupersededAdm.Text = Admx.SupersededAdm
        Dim fillListview = Sub(Control As ListView, Collection As IEnumerable, IdSelector As Func(Of Object, String), NameSelector As Func(Of Object, String))
                               Control.Items.Clear()
                               For Each item In Collection
                                   Dim lsvi = Control.Items.Add(IdSelector(item))
                                   lsvi.Tag = item
                                   lsvi.SubItems.Add(NameSelector(item))
                               Next
                               Control.Columns(1).Width = Control.ClientRectangle.Width - Control.Columns(0).Width - SystemInformation.VerticalScrollBarWidth
                           End Sub
        fillListview(LsvPolicies, Workspace.Policies.Values.Where(Function(p) p.RawPolicy.DefinedIn Is Admx), Function(p As PolicyPlusPolicy) p.RawPolicy.ID, Function(p As PolicyPlusPolicy) p.DisplayName)
        fillListview(LsvCategories, Workspace.FlatCategories.Values.Where(Function(c) c.RawCategory.DefinedIn Is Admx), Function(c As PolicyPlusCategory) c.RawCategory.ID, Function(c As PolicyPlusCategory) c.DisplayName)
        fillListview(LsvProducts, Workspace.FlatProducts.Values.Where(Function(p) p.RawProduct.DefinedIn Is Admx), Function(p As PolicyPlusProduct) p.RawProduct.ID, Function(p As PolicyPlusProduct) p.DisplayName)
        fillListview(LsvSupportDefinitions, Workspace.SupportDefinitions.Values.Where(Function(s) s.RawSupport.DefinedIn Is Admx), Function(s As PolicyPlusSupport) s.RawSupport.ID, Function(s As PolicyPlusSupport) s.DisplayName)
        ShowDialog()
    End Sub
    Private Sub LsvPolicies_DoubleClick(sender As Object, e As EventArgs) Handles LsvPolicies.DoubleClick
        DetailPolicy.PresentDialog(LsvPolicies.SelectedItems(0).Tag)
    End Sub
    Private Sub LsvCategories_DoubleClick(sender As Object, e As EventArgs) Handles LsvCategories.DoubleClick
        DetailCategory.PresentDialog(LsvCategories.SelectedItems(0).Tag)
    End Sub
    Private Sub LsvProducts_DoubleClick(sender As Object, e As EventArgs) Handles LsvProducts.DoubleClick
        DetailProduct.PresentDialog(LsvProducts.SelectedItems(0).Tag)
    End Sub
    Private Sub LsvSupportDefinitions_DoubleClick(sender As Object, e As EventArgs) Handles LsvSupportDefinitions.DoubleClick
        DetailSupport.PresentDialog(LsvSupportDefinitions.SelectedItems(0).Tag)
    End Sub
End Class

File: PolicyPlus/DetailAdmx.cs
using System;
using System.Collections;
using System.Linq;
using System.Windows.Forms;

namespace PolicyPlus
{
    public partial class DetailAdmx
    {
        public DetailAdmx()
        {
            InitializeComponent();
        }
        public void PresentDialog(AdmxFile Admx, AdmxBundle Workspace)
        {
            TextPath.Text = Admx.SourceFile;
            TextNamespace.Text = Admx.AdmxNamespace;
            TextSupersededAdm.Text = Admx.SupersededAdm;
            void fillListview(ListView Control, IEnumerable Collection, Func<object, string> IdSelector, Func<object, string> NameSelector)
            {
                Control.Items.Clear();
                foreach (var item in Collection)
                {
                    var lsvi = Control.Items.Add(IdSelector(item));
                    lsvi.Tag = item;
                    lsvi.SubItems.Add(NameSelector(item));
                }
                Control.Columns[1].Width = Control.ClientRectangle.Width - Control.Columns[0].Width - SystemInformation.VerticalScrollBarWidth;
            };
            fillListview(LsvPolicies, Workspace.Policies.Values.Where(p => ReferenceEquals(p.RawPolicy.DefinedIn, Admx)), (PolicyPlusPolicy p) => p.RawPolicy.ID, (PolicyPlusPolicy p) => p.DisplayName);
            fillListview(LsvCategories, Workspace.FlatCategories.Values.Where(c => ReferenceEquals(c.RawCategory.DefinedIn, Admx)), (PolicyPlusCategory c) => c.RawCategory.ID, (PolicyPlusCategory c) => c.DisplayName);
            fillListview(LsvProducts, Workspace.FlatProducts.Values.Where(p => ReferenceEquals(p.RawProduct.DefinedIn, Admx)), (PolicyPlusProduct p) => p.RawProduct.ID, (PolicyPlusProduct p) => p.DisplayName);
            fillListview(LsvSupportDefinitions, Workspace.SupportDefinitions.Values.Where(s => ReferenceEquals(s.RawSupport.DefinedIn, Admx)), (PolicyPlusSupport s) => s.RawSupport.ID, (PolicyPlusSupport s) => s.DisplayName);
            ShowDialog();
        }
        private void LsvPolicies_DoubleClick(object sender, EventArgs e)
        {
            My.MyProject.Forms.DetailPolicy.PresentDialog((PolicyPlusPolicy)LsvPolicies.SelectedItems[0].Tag);
        }
        private void LsvCategories_DoubleClick(object sender, EventArgs e)
        {
            My.MyProject.Forms.DetailCategory.PresentDialog((PolicyPlusCategory)LsvCategories.SelectedItems[0].Tag);
        }
        private void LsvProducts_DoubleClick(object sender, EventArgs e)
        {
            My.MyProject.Forms.DetailProduct.PresentDialog((PolicyPlusProduct)LsvProducts.SelectedItems[0].Tag);
        }
        private void LsvSupportDefinitions_DoubleClick(object sender, EventArgs e)
        {
            My.MyProject.Forms.DetailSupport.PresentDialog((PolicyPlusSupport)LsvSupportDefinitions.SelectedItems[0].Tag);
        }
    }
}